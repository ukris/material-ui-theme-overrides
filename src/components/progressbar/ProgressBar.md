Standard Component
```jsx
    <ProgressBar
        value={87}
    />
```