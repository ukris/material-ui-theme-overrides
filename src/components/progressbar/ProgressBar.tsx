import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LinearProgress from '@material-ui/core/LinearProgress';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
  },
  bar: {
      height: '15px',
      borderRadius: '2px'
  }
}));

export default function ProgressBar({ value }: { value: number}) {
  const classes = useStyles();

  return (
    <div className={classes.root} title={`${value}%`} data-test="table-progressbar">
      <LinearProgress className={classes.bar} variant="determinate" value={value} />
    </div>
  );
}