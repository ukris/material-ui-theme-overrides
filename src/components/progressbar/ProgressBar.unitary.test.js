import React from 'react'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ProgressBar from './ProgressBar'

configure({ adapter: new Adapter() })


describe('ProgressBar | Non-editable', () => {
    const value = 63
    const component = (<ProgressBar value={value} />)


    it('renders component correctly', () => {
        const wrapper = mount(component)
        const ProgressBar = wrapper.find('[data-test="table-progressbar"]')

        expect(ProgressBar.length).toEqual(1)
        expect(ProgressBar.props().title).toEqual('63%')
        expect(ProgressBar.props().children.props.value).toEqual(63)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
        expect(tree.children[0].props.role).toEqual('progressbar')
        expect(tree.children[0].props['aria-valuenow']).toEqual(63)
    });
})
