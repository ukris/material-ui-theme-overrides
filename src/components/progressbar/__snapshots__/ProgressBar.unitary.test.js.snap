// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar | Non-editable matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-test="table-progressbar"
  title="63%"
>
  <div
    aria-valuenow={63}
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-bar-2 MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
      style={
        Object {
          "transform": "translateX(-37%)",
        }
      }
    />
  </div>
</div>
`;
